src/app/use-cases/status-signature-request.ts

```diff
@@ -0,0 +1,123 @@
+import * as E from "fp-ts/Either";
+import * as t from "io-ts";
+
+import { pipe } from "fp-ts/function";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+import { ProblemDetail } from "@pagopa/handler-kit/lib/problem-detail";
+import { SignatureRequest } from "../../signature-request/signature-request";
+
+export class ActionNotAllowedError extends Error implements ProblemDetail {
+  type = "/problems/action-not-allowed";
+  title = "Action not allowed";
+  status = "400";
+  constructor(public detail: string) {
+    super(detail);
+    this.name = "ActionNotAllowedError";
+  }
+}
+
+export const SignatureRequestAction = t.keyof({
+  UPLOAD_DOCUMENT: null,
+  MARK_AS_READY: null,
+  VALIDATE_DOCUMENT: null,
+  MARK_AS_SIGNED: null,
+});
+export type SignatureRequestAction = t.TypeOf<typeof SignatureRequestAction>;
+
+export const nextStatus =
+  (request: SignatureRequest) =>
+  (
+    action: SignatureRequestAction
+  ): E.Either<ActionNotAllowedError, SignatureRequest> => {
+    switch (request.status) {
+      case "DRAFT":
+        return pipe(action, whenDraft(request));
+      case "WAIT_FOR_ISSUER":
+        return pipe(action, whenWaitForIssuer(request));
+      case "READY":
+        return pipe(action, whenReady(request));
+      case "WAIT_FOR_SIGNATURE":
+        return pipe(action, whenWaitForSignature(request));
+      case "SIGNED":
+        return E.left(
+          new ActionNotAllowedError(
+            "This action is not allowed when signature request is in already SIGNED!"
```

Comment:
I suggest to use quill to correct spelling https://quillbot.com/

Reply:
"This operation is prohibited if the signature request has already been signed."